### PR TITLE
proxy task 11 add post endpoint (retrieve image)

### PR DESCRIPTION
**Pull request by parazit-mc**

### CODE REVIEW

File: controllers/image.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const axios = require('axios');
+const { buildImageUri } = require('../helpers/helpers');
+
+async function roverImageController(req, res) {
+    try {
+        console.log("Authenticating user " + req.body.userId + " " + req.body.userName + ", api key " + req.body.apiKey);
+        const uri = buildImageUri(req.body.apiKey);
+        const response = await axios.get(uri);
```
Comment (mariaYurcheuskaya) [inline controllers/image.js:8]: You can split s controller a little bit, and move the logic to service

File: controllers/image.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        const imageUrl = response.data.latest_photos[0].img_src;
+        res.status(200).send({imageUrl});
+    }
+    catch (error) {
+        console.error(error);
+        res.status(500).json({ error: error, status: 500 });
```
Comment (mariaYurcheuskaya) [inline controllers/image.js:14]: you can use approach with middleware exception handling (you can check in Teams examples)

**Review by mariaYurcheuskaya — APPROVED** [review]

File: controllers/image.js (hunk 1 above)
Comment (parazit-mc) [reply controllers/image.js]: done

File: controllers/image.js (hunk 2 above)
Comment (parazit-mc) [reply controllers/image.js]: done

File: src/services/imageService.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const axios = require('axios');
+const qs = require('qs');
+
+async function getImage(req){
+    console.log("Authenticating user " + req.body.userId + " " + req.body.userName + ", api key " + req.body.apiKey);
```
Comment (plutsenka) [inline src/services/imageService.js:5]: You should use the same quotation marks here, or you can even  do smth like `${placeholder}`

File: src/services/meteorService.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    const queryDate = req.query.date ? new Date(req.query.date).toISOString().split('T')[0] : null;
+    const isHazardous = req.query.isHazardous === 'true';
+    const count = req.query.count ? parseInt(req.query.count) : null;
+    return getMeteorsInfo(meteorsNearby, queryDate, isHazardous, count);
+}
+
+function getMeteorsInfo(meteorsNearby, queryDate, isHazardous, count){
+
+    let meteors =  Object.values(meteorsNearby).flatMap(date=>
+        date.map(m=>({
```
Comment (plutsenka) [inline src/services/meteorService.js:19]: date.map(m => ({

**Pull request merged**